src/liballoc/btree/map.rs

```diff
@@ -2104,6 +2104,32 @@ impl<'a, K: Ord, V> Entry<'a, K, V> {
     }
 }
 
+impl<'a, K: Ord, V: Default> Entry<'a, K, V> {
+    #[unstable(feature = "entry_or_default", issue = "44324")]
+    /// Ensures a value is in the entry by inserting the default value if empty,
+    /// and returns a mutable reference to the value in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::BTreeMap;
```

Comment:
You need to add `#![feature(entry_or_default)]` to the doc test, otherwise it won't pass.

```rust
/// ```
/// #![feature(entry_or_default)]
/// # fn main() {                                   // <-- note that the `fn main` is required
/// use std::collections::BTreeMap;
/// // etc.
/// assert_eq!(map["poneyland"], None);
/// # }
/// ```
```